src/analyzer.rs

```diff
@@ -9,437 +9,503 @@ use std::collections::{BTreeMap, HashMap};
-    let mut res = vec![];
-    analyze_(&schema, "", kind, 0, &mut res)?;
-    Ok(Output(res))
+/// Analyzer provides a method to analyze a schema and generate structs out of it.
+#[derive(Debug, Default)]
+pub struct Analyzer {
+    /// Instructs the analyzer to use k8s_openapi::apimachinery::pkg::apis::meta::v1::Condition
+    /// instead of generating a custom Condition API for each instance.
+    use_k8s_openapi_condition: bool,
 }
 
-/// Scan a schema for structs and members, and recurse to find all structs
-///
-/// schema: root schema / sub schema
-/// current: current key name (or empty string for first call) - must capitalize first letter
-/// stack: stacked concat of kind + current_{n-1} + ... + current (used to create dedup names/types)
-/// level: recursion level (start at 0)
-/// results: multable list of generated structs (not deduplicated)
-fn analyze_(
-    schema: &JSONSchemaProps,
-    current: &str,
-    stack: &str,
-    level: u8,
-    results: &mut Vec<Container>,
-) -> Result<()> {
-    let props = schema.properties.clone().unwrap_or_default();
-    let mut array_recurse_level: HashMap<String, u8> = Default::default();
-
-    // create a Container if we have a container type:
-    //trace!("analyze_ with {} + {}", current, stack);
-    if schema.type_.clone().unwrap_or_default() == "object" {
-        // we can have additionalProperties XOR properties
-        // https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation
-        if let Some(JSONSchemaPropsOrBool::Schema(s)) = schema.additional_properties.as_ref() {
-            let dict_type = s.type_.clone().unwrap_or_default();
-            // object with additionalProperties == map
-            if let Some(extra_props) = &s.properties {
-                // map values is an object with properties
-                debug!("Generating map struct for {} (under {})", current, stack);
-                let c = extract_container(extra_props, stack, &mut array_recurse_level, level, schema)?;
-                results.push(c);
-            } else if !dict_type.is_empty() {
-                warn!("not generating type {} - using {} map", current, dict_type);
-                return Ok(()); // no members here - it'll be inlined
-            }
-        } else {
-            // else, regular properties only
-            debug!("Generating struct for {} (under {})", current, stack);
-            // initial analysis of properties (we do not recurse here, we need to find members first)
-            if props.is_empty() && schema.x_kubernetes_preserve_unknown_fields.unwrap_or(false) {
-                warn!("not generating type {} - using BTreeMap", current);
-                return Ok(());
-            }
-            let c = extract_container(&props, stack, &mut array_recurse_level, level, schema)?;
-            results.push(c);
+impl Analyzer {
+    pub fn new(use_k8s_openapi_condition: bool) -> Self {
```

Comment:
can we create an `analyzer::Config` containing this bool, and then pass that around &cfg within analyze_ and child fns? the places where you pass the bool are getting quite big